WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/FeeLineDetailsViewModel.swift

```diff
@@ -0,0 +1,118 @@
+import SwiftUI
+import struct Yosemite.OrderFeeLine
+
+class FeeLineDetailsViewModel: ObservableObject {
+
+    /// Closure to be invoked when the fee line is updated.
+    ///
+    var didSelectSave: ((OrderFeeLine?) -> Void)
+
+    /// Helper to format price field input.
+    ///
+    private let priceFieldFormatter: PriceFieldFormatter
+
+    /// Formatted amount to display. When empty displays a placeholder value.
+    ///
+    var formattedAmount: String {
+        priceFieldFormatter.formattedAmount
+    }
+
+    /// Stores the amount(unformatted) entered by the merchant.
+    ///
+    @Published var amount: String = "" {
+        didSet {
+            guard amount != oldValue else { return }
+            amount = priceFieldFormatter.formatAmount(amount)
+        }
+    }
+
+    /// Defines the amount text color.
+    ///
+    var amountTextColor: UIColor {
+        amount.isEmpty ? .textPlaceholder : .text
+    }
+
+    /// The base amount (items + shipping) to apply percentage fee on.
+    ///
+    private let baseAmountForPercentage: Decimal
+
+    /// The initial fee amount.
+    ///
+    private let initialAmount: Decimal
+
+    /// Returns true when existing fee line is edited.
+    ///
+    let isExistingFeeLine: Bool
+
+    /// Returns true when there are no valid pending changes.
+    ///
+    var shouldDisableDoneButton: Bool {
+        guard let amountDecimal = priceFieldFormatter.amountDecimal, amountDecimal > .zero else {
+            return true
+        }
+        let finalAmount = feeType == .percentage ? amountDecimal * 0.01 : amountDecimal
```

Comment:
This isn't working quite right for the `.percentage` case. Since we're checking against the initial amount, which is the calculated fixed amount, I think it needs to be this (percentage * base amount):

```suggestion
        let finalAmount = feeType == .percentage ? baseAmountForPercentage * amountDecimal * 0.01 : amountDecimal
```

Reply:
Good catch! Fixed in a59ed90, added test in dc03bcc.